yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,223 @@
+            yield {
+                'id': post_uri,
+                'text': traverse_obj(post, ('record', 'text')),
+                'timestamp': parse_iso8601(traverse_obj(post, ('record', 'createdAt'))),
+                'parent': 'root' if parent_uri == root_uri else parent_uri,
+                'like_count': post.get('likeCount'),
+                'author': traverse_obj(post, ('author', 'displayName')),
+                'author_id': author_did,
+                'author_thumbnail': traverse_obj(post, ('author', 'avatar'), expected_type=url_or_none),
+                'author_url': f'https://bsky.app/profile/{author_handle}',
+                'author_is_uploader': author_did in root_uri,
+            }
+        if replies := thread_node.get('replies'):
+            for reply in replies:
+                yield from self.traverse_replies(reply, root_uri)
+        if parent := thread_node.get('parent'):
+            yield from self.traverse_replies(parent, root_uri)
+
+    def _real_extract(self, url):
+        handle, video_id = self._match_valid_url(url).groups()
+        did = handle if handle.startswith('did:') else self._download_json(
+            'https://public.api.bsky.app/xrpc/com.atproto.identity.resolveHandle',
+            video_id, query={'handle': handle}).get('did')
+
+        getcomments = self.get_param('getcomments', False)
+        meta = self._download_json(
+            'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread',
+            video_id, headers={'Content-Type': 'application/json'}, query={
+                'uri': f'at://{did}/app.bsky.feed.post/{video_id}',
+                'depth': 1000 if getcomments else 0,
+                'parentHeight': 1000 if getcomments else 0,
+            }).get('thread')
+        post, record_embed = meta.get('post'), traverse_obj(meta, ('post', 'record', 'embed'))
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            traverse_obj(post, ('embed', 'playlist'), ('embed', 'media', 'playlist')),
+            video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False,
+            note='Downloading HD m3u8 information', errnote='Unable to download HD m3u8 information')
+        blob_cid = traverse_obj(record_embed, ('video', 'ref', '$link'), ('video', 'cid'),
+                                ('media', 'video', 'ref', '$link'), ('media', 'video', 'cid'))
+        if blob_cid:
+            formats.append({
+                'format_id': 'blob',
+                'url': f'https://bsky.social/xrpc/com.atproto.sync.getBlob?did={did}&cid={blob_cid}',
+                'ext': mimetype2ext(traverse_obj(record_embed, ('video', 'mimeType')), 'mp4'),
+                'width': traverse_obj(record_embed, ('aspectRatio', 'width'), expected_type=int_or_none),
+                'height': traverse_obj(record_embed, ('aspectRatio', 'height'), expected_type=int_or_none),
+                'filesize': traverse_obj(record_embed, ('video', 'size'), expected_type=int_or_none),
```

Comment:
You can combine `traverse_obj` calls. Please also do this for the info dict below.
```suggestion
                **traverse_obj(record_embed, {
                    'ext': ('video', 'mimeType', {mimetype2ext}),
                    'width': ('aspectRatio', 'width', {int_or_none}),
                    'height': ('aspectRatio', 'height', {int_or_none}),
                    'filesize': ('video', 'size', {int_or_none}),
                }),
```